### PR TITLE
[Example] Add DeepLab Example

### PR DESCRIPTION
**Pull request by pinzhenx**
Fix https://github.com/intel/webml-polyfill/issues/313

### CODE REVIEW

File: examples/deeplab/DeepLabImporter.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
+class DeepLabImporter {
```
Comment (huningxin) [inline examples/deeplab/DeepLabImporter.js]: Are we able to merge it into the TFLite model importer?

File: examples/deeplab/utils.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    let imHeight = image.naturalHeight | image.videoHeight;
+    // assume width == height
+    let resizeRatio = Math.max(Math.max(imWidth, imHeight) / width, 1);
+    let scaledWidth = Math.floor(imWidth / resizeRatio);
+    let scaledHeight = Math.floor(imHeight / resizeRatio);
+    let ctx = canvas.getContext('2d');
+    timestamp.push(performance.now());
+    ctx.drawImage(image, 0, 0, scaledWidth, scaledHeight);
+    timestamp.push(performance.now());
+    // // padding (replicate)
```
Comment (huningxin) [inline examples/deeplab/utils.js]: if we don't need padding code, please remove them.

**Comment by huningxin** [comment]
Thanks @pinzhenx ! Please avoid to uploading the model file itself in this PR. Instead, please write a README.md in model folder to instruct how to convert the models and the URLs of pre-converted models.

**Comment by huningxin** [comment]
@pinzhenx , for features, can we demonstrate the background removal and replacement with sample codes? Perhaps you need WebGL shader code to do that efficiently.

**Comment by huningxin** [comment]
@pinzhenx , good progress, thanks! Is it possible to keep the segmentation in color feature (maybe re-implement in WebGL?) besides the background removal/blur/replacement features?

**Comment by pinzhenx** [comment]
Sure. I will turn it into a multipurpose demo.

**Comment by huningxin** [comment]
> Sure. I will turn it into a multipurpose demo.

awesome!

**Comment by pinzhenx** [comment]
I have documented in `README` how to generate the `.tflite` models.
There is a bug in TOCO preventing us from directly generating a model with native dilated convolution. So I haven't put the steps of converting a 'dilated' model in it. I have already filed [an issue](https://github.com/tensorflow/tensorflow/issues/24337) to the tensorflow community.

**Comment by pinzhenx** [comment]
I have updated the guided filter on my demo site. It looks like it only improves the 224 results and doesn't do much for the 513 version. Next, I will focus on the model training.

**Comment by huningxin** [comment]
@pinzhenx , I didn't find where you deal with dilation rates in dilated model. Could you please update the code to use newly added `ATROUS_CONV_2D` and `ATROUS_DEPTHWISE_CONV_2D`? Thanks!

**Comment by pinzhenx** [comment]
@huningxin I haven't yet checked in those lines regarding dilation, since other examples could go wrong before. I will update the code of TFLiteImpoter in next commit, and leave the implementation of WebGL to @GreyZzzzzzXh.

**Comment by pinzhenx** [comment]
The following two results are taken from WebGL backend on 224 model. The inference time is dominated by data synchronization between CPU and GPU.
screen shot 2018-12-24 at 7 54 28 pm
screen shot 2018-12-24 at 8 00 07 pm

**Comment by pinzhenx** [comment]
prototype of WebGL with argmax is available on https://pinzhenx.github.io/webml-demo/examples/deeplab_argmax/camera

**Comment by pinzhenx** [comment]
@huningxin This PR is all been done. PTAL, thanks!

**Comment by huningxin** [comment]
Good job! 

Two comments:
1. if `examples/deeplab/img/riding.jpg` is not used, please remove it
2. could you please add the instructions of how to generate smaller sized model (224x224 and 321x321)

File: examples/deeplab/src/DrawOutputs.js [hunk 3]
```diff
@@ -0,0 +1,886 @@
...
+
+    this.shaders = {};
+
+    this._segMap = null;
+    this._predictions = null;
+    this._clippedSize = [224, 224];
+    this._imageSource = null;
+
+    // workaround for lack of `align_corner` in `RESIZE_BILINEAR`
+    this._correctionFactor = 0.99;
```
Comment (huningxin) [inline examples/deeplab/src/DrawOutputs.js]: Please remove this workaround and file an issue to support `align_corner` in WebNN API.

**Comment by huningxin** [comment]
@pinzhenx , what's the reason to update `examples/util/tflite/schema/schema_generated.js`? Did you test it with other examples?

**Comment by pinzhenx** [comment]
@huningxin The current schema doesn't support depthwise conv with dilation. All tflite based examples work well with this new schema.

**Comment by pinzhenx** [comment]
@huningxin I've made some changes to this PR. PTAL. Thanks.

**Comment by huningxin** [comment]
Looks good. Thanks for this PR!

**Pull request merged**